src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -141,7 +141,17 @@ private async Task<bool> IsEnabledAsync<TContext>(string feature, TContext appCo
                             continue;
                         }
 
-                        IFeatureFilterMetadata filter = GetFeatureFilterMetadata(featureFilterConfiguration.Name);
+                        IFeatureFilterMetadata filter;
+
+                        if (useAppContext)
```

Comment:
If we're now saying a call to `IsEnabledAsync("name", Context);` will use "name" even if it has no context, then I don't think the `useAppContext` bool needs to exist anymore. 

Just use contextual filter if its defined for the context, otherwise use the no contextual filter.

Reply:
Context may be null. I think we still need to distinguish the case that users intend to pass null as the context.